boa/src/exec/identifier/mod.rs

```diff
@@ -1,21 +1,16 @@
 use super::{Executable, Interpreter};
-use crate::{
-    builtins::value::{ResultValue, Value, ValueData},
-    syntax::ast::node::identifier::Identifier,
-};
+use crate::{builtins::value::ResultValue, syntax::ast::node::identifier::Identifier};
 
 impl Executable for Identifier {
     fn run(&self, interpreter: &mut Interpreter) -> ResultValue {
-        let reference = resolve_binding(interpreter, self.as_ref());
-        match reference.data() {
-            ValueData::Undefined => Err(interpreter
-                .throw_reference_error(self.as_ref())
-                .expect_err("throw_reference_error() must return an error")),
-            _ => Ok(reference),
-        }
+        interpreter
+            .realm()
+            .environment
+            .get_binding_value(self.as_ref())
+            .ok_or_else(|| {
+                interpreter
+                    .throw_reference_error(self.as_ref())
+                    .expect_err("throw_reference_error() must return an error")
```

Comment:
Here we can use the `.construct_reference_error()` so we don't have to call `.expect_err()`
```suggestion
                interpreter.construct_reference_error(self.as_ref())
```